File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> valueToIndex =  new Dictionary<int,int>();
        for (int i = 0; i < nums.Length; i++)
        {
            valueToIndex[nums[i]] = i;
        }

        for (int i = 0; i < nums.Length; i++)
        {
            if (valueToIndex.ContainsKey(target - nums[i]) && valueToIndex[target - nums[i]] != i)
            {
                return [i, valueToIndex[target - nums[i]]];
            }
        }
        return [0,0];
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.java
class Solution {
    public boolean canJump(int[] nums) {
        int cur = nums[0];
        for(int i = 0; i < nums.length; i++){
            cur--;
            if(nums[i] > cur){
                cur = nums[i];
            }
            if(cur <= 0 && i != nums.length-1){
                return false;
            }
        }
        return true;
    }
}
